skillmap/src/App.tsx

```diff
@@ -407,6 +419,12 @@ class AppImpl extends React.Component<AppProps, AppState> {
                 <MakeCodeFrame forcelang={forcelang} onWorkspaceReady={this.onMakeCodeFrameLoaded}/>
                 <AppModal />
                 <UserProfile />
+                <LanguageSelector
+                    visible={this.props.showSelectLanguage}
```

Comment:
I'm wondering why we would pass `visible` as a prop. Why not make this a conditional div instead? 
Like `{ this.props.showSelectLanguage && <LanguageSelector`

Reply:
tl;dr: because that's what UserProfile does and I wanted to be consistent, but I'm open to switching it if you feel strongly about doing it the other way

longer:
Mostly I was just following the pattern I saw in similar controls on this page (namely UserProfile above), where you include the element and manage visibility from inside the control itself. But I don't feel strongly either way.

The reason I actually have to pass it in here (whereas UserProfile doesn't have the parameter), is because I'm not using the React Redux pattern UserProfile uses (with the whole `connect` method, etc...). And the reason I'm not using redux is because LanguageSelector is in react-common, where I don't see it getting used much. I don't know much about it TBH, but I get the impression redux would have coupled the control pretty tightly to some skillmap state and made it difficult to reuse in other areas of the code (like multiplayer).

Reply:
Hmm yea I don't feel strongly, I just thought this way might be simpler, but it also makes sense to keep things consistent.

Reply:
The more I think about it though, it does seem kind of unnecessary to have it as a prop if there's no other need for it within the component.

Reply:
Moved it out, as you suggested